### DISCUSSION TITLE
Get 401 error when trying to upload photos in Lychee 5.0.2

### DISCUSSION THREAD

**Discussion by jakeanderson1** [LycheeOrg/Lychee#2212] [Q&A]
### Detailed description of the problem [REQUIRED]
I recently updated the docker image for Lychee and started seeing issues with the upload behavior. I have attempted to disable LiveWire but I don't think I've done so successfully. Some other notes are that I am running it in kubernetes and behind an nginx ingress, which did not have issues with 4.x but now seems to be preventing the file upload in the newest version.

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
1. Log in
2. Click the plus icon in top right and select Upload Photo
3. Drag and drop a file into the dialog
4. Upload fails, shows 401 in network console

**Screenshots**
<img width="514" alt="image" src="https://github.com/LycheeOrg/Lychee/assets/10400771/73c604bb-b1b6-44c5-8e4b-46efdbdcb5d9">

Here is the stack trace from the response of /livewire/upload-file?expires=1704389775&signature=21fe80477a84a8229c51d2aad87203f2bd466394f88cbaa15616f8cf8d73b99b
{
    "message": "",
    "exception": "Symfony\\Component\\HttpKernel\\Exception\\HttpException",
    "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Application.php",
    "line": 1248,
    "trace": [
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php",
            "line": 45,
            "function": "abort",
            "class": "Illuminate\\Foundation\\Application",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php",
            "line": 86,
            "function": "abort"
        },
        {
            "file": "/var/www/html/Lychee/vendor/livewire/livewire/src/Features/SupportFileUploads/FileUploadController.php",
            "line": 27,
            "function": "abort_unless"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php",
            "line": 46,
            "function": "handle",
            "class": "Livewire\\Features\\SupportFileUploads\\FileUploadController",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
            "line": 259,
            "function": "dispatch",
            "class": "Illuminate\\Routing\\ControllerDispatcher",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
            "line": 205,
            "function": "runController",
            "class": "Illuminate\\Routing\\Route",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 806,
            "function": "run",
            "class": "Illuminate\\Routing\\Route",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 144,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/QueryStringFixer.php",
            "line": 40,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\QueryStringFixer",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/DisableCSP.php",
            "line": 55,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\DisableCSP",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php",
            "line": 50,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Routing\\Middleware\\SubstituteBindings",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php",
            "line": 78,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/VerifyCsrfToken.php",
            "line": 44,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\VerifyCsrfToken",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php",
            "line": 60,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Session\\Middleware\\AuthenticateSession",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php",
            "line": 159,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php",
            "line": 90,
            "function": "handleRequest",
            "class": "Illuminate\\Routing\\Middleware\\ThrottleRequests",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Routing\\Middleware\\ThrottleRequests",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php",
            "line": 49,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\View\\Middleware\\ShareErrorsFromSession",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php",
            "line": 121,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php",
            "line": 64,
            "function": "handleStatefulRequest",
            "class": "Illuminate\\Session\\Middleware\\StartSession",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Session\\Middleware\\StartSession",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php",
            "line": 37,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php",
            "line": 67,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Cookie\\Middleware\\EncryptCookies",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/AcceptContentType.php",
            "line": 50,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\AcceptContentType",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/AdminUserStatus.php",
            "line": 55,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\AdminUserStatus",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/InstallationStatus.php",
            "line": 55,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\InstallationStatus",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 119,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 805,
            "function": "then",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 784,
            "function": "runRouteWithinStack",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 748,
            "function": "runRoute",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 737,
            "function": "dispatchToRoute",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 200,
            "function": "dispatch",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 144,
            "function": "Illuminate\\Foundation\\Http\\{closure}",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/livewire/livewire/src/Features/SupportDisablingBackButtonCache/DisableBackButtonCacheMiddleware.php",
            "line": 19,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Livewire\\Features\\SupportDisablingBackButtonCache\\DisableBackButtonCacheMiddleware",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/bepsvpt/secure-headers/src/SecureHeadersMiddleware.php",
            "line": 20,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Bepsvpt\\SecureHeaders\\SecureHeadersMiddleware",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
            "line": 21,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php",
            "line": 31,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
            "line": 21,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php",
            "line": 40,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\TrimStrings",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php",
            "line": 27,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php",
            "line": 99,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php",
            "line": 39,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "Illuminate\\Http\\Middleware\\TrustProxies",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/app/Http/Middleware/FixStatusCode.php",
            "line": 30,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 183,
            "function": "handle",
            "class": "App\\Http\\Middleware\\FixStatusCode",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 119,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 175,
            "function": "then",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 144,
            "function": "sendRequestThroughRouter",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        },
        {
            "file": "/var/www/html/Lychee/public/index.php",
            "line": 73,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        }
    ],
    "previous_exception": null
}
*If applicable, add screenshots to help explain your problem.*

### Output of the diagnostics  [REQUIRED]
*(Settings => Diagnostics or https://example.com/Diagnostics or `php artisan lychee:diagnostics`)*
Note on the diagnostics below, I suspect the APP_URL value I have is part of the issue but could not set it to anything that would resolve this error. If I set it to anything other than https://mydomain.com then none of the existing images load because it tries to do them with either http or a bad path.

 Diagnostics
    -----------
        Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
        Error: APP_URL does not match the current url. This will break WebAuthn authentication.
        Error: APP_URL does not match the current url. This will prevent images to be properly displayed.
        Info: Latest version of PHP is 8.3
        Warning: git (software) is not available.
         Foreign key: access_permissions.user_id     → users.id             : CASCADE
         Foreign key: albums.parent_id               → albums.id            : RESTRICT
         Foreign key: albums.id                      → base_albums.id       : RESTRICT
         Foreign key: albums.cover_id                → photos.id            : CASCADE
         Foreign key: base_albums.owner_id           → users.id             : RESTRICT
         Foreign key: jobs_history.owner_id          → users.id             : RESTRICT
         Foreign key: photos.album_id                → albums.id            : RESTRICT
         Foreign key: photos.owner_id                → users.id             : RESTRICT
         Foreign key: size_variants.photo_id         → photos.id            : RESTRICT
         Foreign key: sym_links.size_variant_id      → size_variants.id     : RESTRICT
         Foreign key: tag_albums.id                  → base_albums.id       : RESTRICT

### Browser and system
Brave/Chromium MacOS Sonoma 14.2.1

**Comment by ildyria** [comment]
1. are you using Lychee in a directory structure ? e.g. `https://example.com/myLychee/`
2. For the http, you can force https back with `APP_FORCE_HTTPS=true` in your `.env` but a better way is to also forward/force the https protocol in your proxy.

**Comment by jakeanderson1** [comment]
Good questions, no it's not in a directory structure, it's at the root. I tried `APP_FORCE_HTTPS=true` and that did not seem to help. When you say forward https do you mean to enable https within Lychee somehow? Would that also allow me to embed the generated cert into Lychee? Currently I am terminating SSL at the proxy, and then traffic from the ingress to Lychee is over http, which seems to be at the root of the issue. It sees the url from the request as `https://mydomainname.com/` and throws an error because it expects `http://mydomainname.com` instead, which is where Lychee is configured to serve traffic.

**Comment by ildyria** [comment]
I was more referring to this: https://github.com/LycheeOrg/Lychee/discussions/2156

**Comment by jakeanderson1** [comment]
Ah, brilliant! I did not see that, thank you. Let me try putting the right attributes on the ingress so this header is set and will close it if that works.

**Comment by jakeanderson1** [comment]
I tried the headers recommended on the other issue and it's not quite working yet. Here is some debugging output of the value of the headers on a request but the diagnostics page is still reporting an issue with the APP_URL and the image uploads are still failing. The values are redacted but otherwise accurate with what Lychee is seeing.

{
        "accept-language": [
                "en-US,en;q=0.9"
        ],
        "accept-encoding": [
                "gzip, deflate, br"
        ],
        "sec-fetch-dest": [
                "empty"
        ],
        "sec-fetch-mode": [
                "cors"
        ],
        "sec-fetch-site": [
                "same-origin"
        ],
        "origin": [
                "https://mydomainname.com"
        ],
        "sec-gpc": [
                "1"
        ],
        "accept": [
                "*/*"
        ],
        "sec-ch-ua-platform": [
                "\"macOS\""
        ],
        "x-livewire": [
                ""
        ],
        "user-agent": [
                "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
        ],
        "sec-ch-ua-mobile": [
                "?0"
        ],
        "content-type": [
                "application/json"
        ],
        "sec-ch-ua": [
                "\"Not_A Brand\";v=\"8\", \"Chromium\";v=\"120\", \"Brave\";v=\"120\""
        ],
        "content-length": [
                "853"
        ],
        "x-scheme": [
                "https"
        ],
        "x-forwarded-scheme": [
                "https"
        ],
        "x-forwarded-proto": [
                "https"
        ],
        "x-forwarded-port": [
                "443"
        ],
        "x-forwarded-host": [
                "mydomainname.com"
        ],
        "x-forwarded-for": [
                "REDACTED"
        ],
        "x-real-ip": [
                "REDACTED"
        ],
        "x-request-id": [
                "aba449e5a9d1845846017553d1680644"
        ],
        "host": [
                "mydomainname.com"
        ]
}

**Comment by mlapaglia** [comment]
i was having this same problem. i switched the linuxserver lychee image and it started working.

**Comment by jakeanderson1** [comment]
Do you know what the difference is from the official LycheeOrg image? Are there any issues migrating between them, or moving back if I run into problems with the linuxserver image? I will try it but just wanted to check first. Thanks for the recommendation 👍

**Comment by jortgies** [comment]
Same problem running the official docker image behind traefik.

Fixed it by correctly setting the `TRUSTED_PROXIES` in the `.env`-File.

**Comment by jakeanderson1** [comment]
I tried setting `TRUSTED_PROXIES` to my proxy IP but that didn't resolve the issue. It still seems to be a mismatch of http vs https but I can't figure out the right combination of settings for `APP_URL` and `APP_FORCE_HTTPS`. Still have not yet tried the linuxserver image.

**Comment by ildyria** [comment]
if you are accessing your website from HTTPS, set APP_URL to your https url.
You will also need to set up the TRUSTED_PROXIES.

However in your case, my guess is that you are not having the forwarding the https proto in your reverse proxy.

In my apache reverse proxy I have:
```
RequestHeader set X-Forwarded-Proto https
RequestHeader set X-Forwarded-SSL expr=%{HTTPS}
```
There should be similar setting in yours.

**Reply by jakeanderson1** [reply]
Right, if I don't set HTTPS in the APP_URL, all of the image links are broken, and then the diagnostics page complains because the request URL(which is HTTP) doesn't match the APP_URL. So you're suggesting that instead of terminating SSL at the nginx ingress I instead actually forward it to Lychee. I'll have to understand how to set up and renew the cert within the Lychee container in order to do that. 

Perhaps it's also worth asking the question: Is the discrepancy in APP_URL vs the apparent request URL causing my issues with upload, or is that possibly unrelated?

**Comment by wisplite** [comment]
I am experiencing the same issue putting the Lychee docker container behind Caddy reverse proxy, though the log for lychee simply shows the following:
```
Illuminate\Foundation\Application::abort:1389 
App\Http\Middleware\QueryStringFixer::handle:40 ; caused by
```

Any help will be greatly appreciated. My APP_URL is set to my https URL and APP_FORCE_HTTPS is set to true, because if I set it to false it always makes requests over HTTP which Firefox blocks for obvious security reasons. Caddy automatically forwards all headers sent by the client, so none of them should need to be set manually, though I did try setting them manually and nothing changed.

**Reply by wisplite** [reply]
Adding onto this, I fixed it by setting the TRUSTED_PROXIES variable to "*", to trust all proxies. I know this isn't best practice security-wise, but the proxy is running on the same machine as the Lychee container, and the only ports that are exposed go to the proxy, so it should be fine. I suspect there is something weird going on with Docker, from my experience, handling networking between containers and the host is a nightmare.

**Reply by ildyria** [reply]
It is not per say with docker but with proxies themselves.
It is the same if you are running Lychee behind cloudfare as a proxy.

If I am not wrong, ideally you would want the ip of the proxy to be set in TRUSTED_PROXIES.
Also did you set your forward proto for https in your caddy ?

I am using traefic and had the same issue as you. :)

**Reply by wisplite** [reply]
I did set the X-Forwarded-Proto to https, though Caddy should have set that automatically anyway (that's kinda caddy's whole thing, it's supposed to be practically invisible and require little to no setup).

I am running everything behind both Caddy and the Cloudflare proxy though, as I want the DDoS protection that Cloudflare provides. I tried setting the trusted IP to both localhost and 172.17.0.1 (which is what Lychee logged as me connecting through), but nothing I put there seemed to work. Setting it to accept connections from all proxies seems to work perfectly.
